coordinator/src/notification_service.rs

```diff
@@ -130,3 +144,205 @@ async fn send_notification<'a>(
     tracing::debug!("Sent: {:?}", response);
     Ok(())
 }
+
+/// Send notifications to users with positions that are about to expire or have
+/// just expired
+pub async fn send_expiry_notifications_if_applicable(
```

Comment:
nit: reading the code from top to down I would have expected that function to come after `query_and_send_position_notifications` and certainly the tests at the very end of the file. 🙃

Reply:
moved `query_and_send_position_notifications` above this function.